### PR TITLE
docs(errors): Update error class docstrings per the latest RFCs

### LINKED ISSUE #595: Update docstrings for HTTP error classes according to RFC 7231
RFC 7231 clarifies and refines the use of various status codes. Update the docstrings to suite.

### PR DESCRIPTION
**Pull request by kgriffs**
Fixes #595

### CODE REVIEW

File: falcon/errors.py [hunk 1]
```diff
@@ -46,16 +46,46 @@ def on_get(self, req, resp):
...
+
+    (See also: RFC 7231, Section 6.5.1)
 
     Args:
         title (str): Error title (e.g., 'TTL Out of Range').
         description (str): Human-friendly description of the error, along with
             a helpful suggestion or two.
-        kwargs (optional): Same as for ``HTTPError``.
 
+    Keyword Args:
```
Comment (kgriffs) [inline falcon/errors.py]: I also added these sections to be more clear which kwargs can be used, and which are required or already filled in.

**Comment by fxfitz** [comment]
LGTM

**Comment by kgriffs** [comment]
Found a couple places where an arg should be moved to the Keyword Args section, and one class had duplicate title and description args in the docstring.

**Comment by jmvrbanac** [comment]
:+1:

**Pull request merged**